engine/src/main/java/org/hibernate/search/engine/common/impl/SearchIntegrationPartialBuildStateImpl.java

```diff
@@ -164,11 +172,30 @@ public SearchIntegration finalizeIntegration() {
 			}
 			failureCollector.checkNoFailure();
 
+			// Pre-Start indexes
+			for ( Map.Entry<String, IndexManagerNonStartedState> entry : nonStartedIndexManagers.entrySet() ) {
+				SavedState savedState = null;
+				if ( previousIntegration.isPresent() ) {
+					Map<String, SavedState> indexManagersStates = previousIntegration.get().saveForRestart()
+							.get( INDEX_MANAGERS_KEY ).orElse( Collections.emptyMap() );
+
+					savedState = indexManagersStates.get( entry.getKey() );
```

Comment:
I think that theoretically there wouldn't be values that are not reused by the new search integration.
But sure, it is better to introduce such a mechanism to prevent any possible leaks in order to support a more generic case.

Reply:
Yes: it doesn't happen, until it something change in Infinispan for some unrelated feature, and now it does happen, and then we have a leak, and we won't notice it until it's crippling some real-world application... Better safe than sorry.